frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/BaseProjectDependentAction.java

```diff
@@ -0,0 +1,69 @@
+package com.virtuslab.gitmachete.frontend.actions.base;
+
+import com.intellij.openapi.actionSystem.AnActionEvent;
+import com.intellij.openapi.project.DumbAwareAction;
+import com.intellij.openapi.project.Project;
+import git4idea.repo.GitRepository;
+import io.vavr.control.Option;
+import org.checkerframework.checker.guieffect.qual.UIEffect;
+
+import com.virtuslab.branchlayout.api.readwrite.IBranchLayoutWriter;
+import com.virtuslab.gitmachete.frontend.actions.expectedkeys.IWithLogger;
+import com.virtuslab.gitmachete.frontend.ui.api.table.BaseEnhancedGraphTable;
+import com.virtuslab.gitmachete.frontend.ui.providerservice.BranchLayoutWriterProvider;
+import com.virtuslab.gitmachete.frontend.ui.providerservice.GraphTableProvider;
+import com.virtuslab.gitmachete.frontend.ui.providerservice.SelectedGitRepositoryProvider;
+
+public abstract class BaseProjectDependentAction extends DumbAwareAction implements IWithLogger {
+  @Override
+  @UIEffect
+  public final void update(AnActionEvent anActionEvent) {
+    super.update(anActionEvent);
+
+    var maybeProject = anActionEvent.getProject();
+    var presentation = anActionEvent.getPresentation();
+    if (maybeProject == null) {
+      presentation.setEnabledAndVisible(false);
+    } else {
+      presentation.setEnabledAndVisible(true);
+      onUpdate(anActionEvent);
+    }
+  }
+
+  /**
+   * If overridden, {@code super.onUpdate(anActionEvent)} should always be called in the first line of overriding method.
+   * In addition, in this method we should use getters WITHOUT logging like
+   * {@code IExpectsKeyGitMacheteRepository#getGitMacheteRepositorySnapshotWithoutLogging}
+   *
+   * @param anActionEvent an action event
+   */
+  @UIEffect
+  protected abstract void onUpdate(AnActionEvent anActionEvent);
```

Comment:
Because in many actions this method is just empty, so maybe get rid of `abstract` here and make it empty by default?

Reply:
Good idea, done